pkg/storage/bucket/object_client_adapter.go

```diff
@@ -103,26 +103,40 @@ func (o *ObjectClientAdapter) List(ctx context.Context, prefix, delimiter string
 
 	// If delimiter is empty we want to list all files
 	if delimiter == "" {
-		iterParams = append(iterParams, objstore.WithRecursiveIter)
+		iterParams = append(iterParams, objstore.WithRecursiveIter())
 	}
 
-	err := o.bucket.Iter(ctx, prefix, func(objectKey string) error {
+	supportUpdatedAt := objstore.ValidateIterOptions(o.bucket.SupportedIterOptions(), objstore.WithUpdatedAt()) == nil
```

Comment:
i don't like how this validation method turned out  😓 i think alternatively we could use
```
slices.Contains(o.bucket.SupportedIterOptions(), objstore.UpdatedAt)
```